src/Workspaces/VisualBasic/Portable/CodeGeneration/VisualBasicSyntaxGenerator.vb

```diff
@@ -2366,8 +2366,15 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.CodeGeneration
             Dim isDefault As Boolean
             SyntaxFacts.GetAccessibilityAndModifiers(tokens, acc, currentMods, isDefault)
 
-            If (currentMods <> modifiers) Then
-                Dim newTokens = GetModifierList(acc, modifiers And GetAllowedModifiers(declaration.Kind), GetDeclarationKind(declaration), isDefault)
+            If currentMods <> modifiers Then
+                Dim declarationKind = GetDeclarationKind(declaration)
+                If declarationKind = DeclarationKind.None AndAlso declaration.IsKind(SyntaxKind.ClassStatement) Then
```

Comment:
i don't get it.  this is directly making a change purely for classes.  i dont' see why we wouldn't need the same for structs/interfaces/modules here.  

And, like i said before, i thinik the bug is that hte caller is passing ina  class statement, it should pass in a classblock.

Reply:
> i thinik the bug is that hte caller is passing ina class statement, it should pass in a classblock.

I believe API consumers won't expect this at all.

---

> i don't get it. this is directly making a change purely for classes. i dont' see why we wouldn't need the same for structs/interfaces/modules here.

Because it literally doesn't have any effect. If you look at the following method:

https://github.com/dotnet/roslyn/blob/ad0674f544bfd0e5e6110f761ff8cae80a028544/src/Workspaces/VisualBasic/Portable/CodeGeneration/VisualBasicSyntaxGenerator.vb#L2490

Any type declaration other than a class will be treated the same. I could handle all other statements, but it doesn't have any functional/behavioral change.

Reply:
> i thinik the bug is that hte caller is passing ina class statement, it should pass in a classblock.

I believe API consumers won't expect this at all.

---

> i don't get it. this is directly making a change purely for classes. i dont' see why we wouldn't need the same for structs/interfaces/modules here.

Because it literally doesn't have any effect. If you look at the following method:

https://github.com/dotnet/roslyn/blob/ad0674f544bfd0e5e6110f761ff8cae80a028544/src/Workspaces/VisualBasic/Portable/CodeGeneration/VisualBasicSyntaxGenerator.vb#L2490

Any type declaration other than a class will be treated the same. I could handle all other statements, but it doesn't have any functional/behavioral change.

Reply:
To put it another way, there is no way to write a unit test for interface/structure that can fail now and be fixed by handling statements in this code path. `WithModifiers` *already* works correctly for the other declarations.

Reply:
i don't really get it.  

> I believe API consumers won't expect this at all.

I don't have an issue with htat.  It was an explicit decision that we do not try to blur the lines between the statements and blocks.  The block corresponds to the decl, and the api here intends to operate on the decls.

Reply:
this is my approach: https://github.com/dotnet/roslyn/issues/50003

Reply:
@CyrusNajmabadi Well, Part of the disconnect is whether `WithModifiers` should work with statements, or only blocks. It currently works with statements very well, except for the class case (more specifically sealing the class). Anything other than sealing a class works correctly using either a statement or a block. This is one of the reasons I think `WithModifiers` should get modified to work with class statement, even if the original design didn't intend that.

Otherwise, a clear doc comment should make this clear. (and in this case, you can close the PR and #53627)

Reply:
i would then update GetModifierList to put the special logic there.

Reply:
@CyrusNajmabadi The suggested approach is a lot easier to follow and to reason about. Only problem I found is that most callers don't have a relevant SyntaxNode to pass. So I passed `Nothing` for these cases.